src/referencePackageSourceGenerator/ReferencePackageSourceGenerator.proj

```diff
@@ -7,7 +7,7 @@
     <PackagesTargetDirectory>$(RepoRoot)src\referencePackages\src\</PackagesTargetDirectory>
     <!-- The following target frameworks aren't buildable with the current SDK and as they don't contribute to the set of
          source build target verticals, can be safely excluded. -->
-    <ExcludeTargetFrameworks>netcore50;net463;portable-*;xamarin*;monotouch*;monoandroid*;win8;wp8;wpa81</ExcludeTargetFrameworks>
+    <ExcludeTargetFrameworks>netcoreapp3.1;netcore50;net463;portable-*;xamarin*;monotouch*;monoandroid*;win8;wp8;wpa81</ExcludeTargetFrameworks>
```

Comment:
Why do we need to exclude netcoreapp3.1 if a targeting pack for it exists? https://github.com/dotnet/source-build-reference-packages/tree/main/src/targetPacks/ILsrc/microsoft.netcore.app.ref/3.1.0

Reply:
I don't know the history here but perhaps it was because of the underlying issue being fixed here.  That being said, I don't suspect we will need any netcoreapp3.1 TFMs going forward.  I like the idea of filtering the TFM until it is proven they are needed.

Reply:
Based on that I would that think that we should block all netcoreapp* TFMs and remove their targeting packs. But if we would then stumble upon a package that has one of these TFMs in it and we actually require it, we would need to bring the targeting pack back and remove the exclusion filter. That just makes me wonder why do we want to block netcoreapp3.1 but not netcoreapp3.0, netcoreapp2.2, netcoreapp2.1 or netcoreapp2.0?

Reply:
Let's take a step back and look at this holistically and establish a set of principles.  We can log a follow-up issue(s) to achieve these as this is well beyond the scope of what Matt is trying to accomplish here - unblock the generation of net 6.0 SBRPs.

**Principles**
1. The SBRPs in this repo should contain the minimum set of TFMs required by source-build and the product repos.  In the past, the SBRPs contained whatever TFMs the tooling emitted.  Very seldom did folks specify the explicit minimal set of TFMs required by the product.
1. The tooling should support all TFMs required by source-build and the product repos. We know there are always repos that lag behind in what TFMs they target.  If an SBRP is needed for this type of situation, the tooling should be capable of generating ref packs for those TFMs.  Being capable is very different than supporting it "out of the box".

To help achieve the first principal, I propose the tooling filters out a default set of TFMs.  This filter should be ignored when an explicit set of TFMs is specified.  Furthermore, we can create some tooling or modify the prebuilt reporting infra to indicate which TFMs are consumed from the detected prebuilts.  If agreed upon we can log an issue for these and debate what the default TFM filters should be.

Reply:
Thanks for sharing your thoughts on this.

> To help achieve the first principal, I propose the tooling filters out a default set of TFMs. This filter should be ignored when an explicit set of TFMs is specified.

Let me describe how the current include / exclude mechanism works. Target frameworks that are excluded via the [`ExcludedTargetFrameworks`](https://github.com/dotnet/source-build-reference-packages/blob/a848f811a52d75494029e663cd0a799be11744fe/src/referencePackageSourceGenerator/ReferencePackageSourceGenerator.proj#L10) property are always excluded, even if they are explicitly listed in the [`IncludeTargetFrameworks`](https://github.com/dotnet/source-build-reference-packages/blob/a848f811a52d75494029e663cd0a799be11744fe/src/referencePackageSourceGenerator/ReferencePackageSourceGenerator.proj#L71) property. Reason for that is that those TFMs aren't restore or build-able with the SDK used (either because of missing targeting packs or because the SDK just doesn't support targeting the TFM).

If a user invokes the following command `generate.sh --package "System.Drawing.Common,7.0.0,net7.0"` then that means that the `IncludeTargetFrameworks` property is set to `net7.0` and only the net7.0 TFM will be generated.

Let's assume that another package `System.X` with version `x.x.x` contains a `netcore50` compile assembly. If a user invokes `generate.sh --package "System.X,x.x.x,net7.0;netcore50"`, then only the `net7.0` TFM is generated as `netcore50` is excluded as it isn't restore and/or buildable.

Long story short, we are conflating reasons for excluding a TFM. If we want to (by default) exclude TFMs that are supported by the SDK, then we should add the following snippet instead:

```suggestion
    <ExcludeTargetFrameworks>netcore50;net463;portable-*;xamarin*;monotouch*;monoandroid*;win8;wp8;wpa81</ExcludeTargetFrameworks>
    <!-- The following target frameworks should be excluded even though they are supported by the current SDK because there is no need to have them.
         Only exclude them when target frameworks to include aren't provided. -->
    <ExcludeTargetFrameworks Condition="'$(IncludeTargetFrameworks)' == ''">$(ExcludeTargetFrameworks);netcoreapp3.1</ExcludeTargetFrameworks>
```

Reply:
Thanks for filing. Still, for this PR I think @mthalman should take the above suggestion.

Reply:
I've applied the changes suggested in https://github.com/dotnet/source-build-reference-packages/pull/567#discussion_r1132454025.